### ISSUE TITLE
Pivot Table not working with Sandboxed user

### ISSUE THREAD

**Issue by flamber** [metabase/metabase#14969]
**Describe the bug**
Pivot Table not working with Sandboxed user, when the question has joins (implicit or explicit).

**To Reproduce**
1. Admin > People > create user "U1" with attribute `user_id`=`1` and `user_cat`=`Widget`
2. Admin > Permissions > Sample Dataset > sandbox Orders with "User ID"=`user_id` and People with "ID"=`user_id` and Products with "Category"=`user_cat`
![image](https://user-images.githubusercontent.com/1447303/109201223-df04b600-77a1-11eb-8d5e-0ecaa6396887.png)
3. Custom question > Sample Dataset > Orders - join People - summarize Sum of Total by People.Source and Product.Category.
4. Change visualization to Pivot Table - save question as "Q1".
![image](https://user-images.githubusercontent.com/1447303/109201465-2a1ec900-77a2-11eb-8b09-4c5cdd201625.png)
5. Login as "U1" and go to "Q1" - question fails to load - no error in the log, but the browser console has a message (not error):
<details><summary>Full message from the browser console, which includes a stacktrace</summary>

```
19:47:59.059 error 
{…}
​
data: {…}
​​
_status: 500
​​
cause: "Column \"source.ADDRESS\" not found; SQL statement:\nCREATE FORCE VIEW PUBLIC._75 AS\nSELECT\n    \"source\".TOTAL AS TOTAL,\n    \"People - User\".SOURCE AS \"People - User__SOURCE\",\n    \"PRODUCTS__via__PRODUCT_ID\".CATEGORY AS \"PRODUCTS__via__PRODUCT_ID__CATEGORY\",\n    ABS(0) AS \"pivot-grouping\",\n    \"source\".ID AS ID,\n    \"source\".USER_ID AS USER_ID,\n    \"source\".PRODUCT_ID AS PRODUCT_ID,\n    \"source\".SUBTOTAL AS SUBTOTAL,\n    \"source\".TAX AS TAX,\n    \"source\".DISCOUNT AS DISCOUNT,\n    \"source\".CREATED_AT AS CREATED_AT,\n    \"source\".QUANTITY AS QUANTITY,\n    \"source\".ID AS ID_2,\n    \"source\".ADDRESS AS ADDRESS,\n    \"source\".EMAIL AS EMAIL,\n    \"source\".PASSWORD AS PASSWORD,\n    \"source\".NAME AS NAME,\n    \"source\".CITY AS CITY,\n    \"source\".LONGITUDE AS LONGITUDE,\n    \"source\".STATE AS STATE,\n    \"source\".SOURCE AS SOURCE,\n    \"source\".BIRTH_DATE AS BIRTH_DATE,\n    \"source\".ZIP AS ZIP,\n    \"source\".LATITUDE AS LATITUDE,\n    \"source\".CREATED_AT AS CREATED_AT_2,\n    \"People - User\".ID AS \"People - User__ID\",\n    \"source\".CATEGORY AS CATEGORY,\n    \"source\".ID AS ID_3,\n    \"source\".EAN AS EAN,\n    \"source\".TITLE AS TITLE,\n    \"source\".VENDOR AS VENDOR,\n    \"source\".PRICE AS PRICE,\n    \"source\".RATING AS RATING,\n    \"source\".CREATED_AT AS CREATED_AT_3,\n    \"PRODUCTS__via__PRODUCT_ID\".ID AS \"PRODUCTS__via__PRODUCT_ID__ID\"\nFROM (\n    SELECT\n        PUBLIC.ORDERS.ID AS ID,\n        PUBLIC.ORDERS.USER_ID AS USER_ID,\n        PUBLIC.ORDERS.PRODUCT_ID AS PRODUCT_ID,\n        PUBLIC.ORDERS.SUBTOTAL AS SUBTOTAL,\n        PUBLIC.ORDERS.TAX AS TAX,\n        PUBLIC.ORDERS.TOTAL AS TOTAL,\n        PUBLIC.ORDERS.DISCOUNT AS DISCOUNT,\n        PUBLIC.ORDERS.CREATED_AT AS CREATED_AT,\n        PUBLIC.ORDERS.QUANTITY AS QUANTITY\n    FROM PUBLIC.ORDERS\n    WHERE PUBLIC.ORDERS.USER_ID = 1\n) \"source\"\nLEFT OUTER JOIN (\n    SELECT\n        PUBLIC.PEOPLE.ID AS ID,\n        PUBLIC.PEOPLE.ADDRESS AS ADDRESS,\n        PUBLIC.PEOPLE.EMAIL AS EMAIL,\n        PUBLIC.PEOPLE.PASSWORD AS PASSWORD,\n        PUBLIC.PEOPLE.NAME AS NAME,\n        PUBLIC.PEOPLE.CITY AS CITY,\n        PUBLIC.PEOPLE.LONGITUDE AS LONGITUDE,\n        PUBLIC.PEOPLE.STATE AS STATE,\n        PUBLIC.PEOPLE.SOURCE AS SOURCE,\n        PUBLIC.PEOPLE.BIRTH_DATE AS BIRTH_DATE,\n        PUBLIC.PEOPLE.ZIP AS ZIP,\n        PUBLIC.PEOPLE.LATITUDE AS LATITUDE,\n        PUBLIC.PEOPLE.CREATED_AT AS CREATED_AT\n    FROM PUBLIC.PEOPLE\n    WHERE PUBLIC.PEOPLE.ID = 1\n) \"People - User\"\n    ON \"source\".USER_ID = \"People - User\".ID\nLEFT OUTER JOIN (\n    SELECT\n        PUBLIC.PRODUCTS.ID AS ID,\n        PUBLIC.PRODUCTS.EAN AS EAN,\n        PUBLIC.PRODUCTS.TITLE AS TITLE,\n        PUBLIC.PRODUCTS.CATEGORY AS CATEGORY,\n        PUBLIC.PRODUCTS.VENDOR AS VENDOR,\n        PUBLIC.PRODUCTS.PRICE AS PRICE,\n        PUBLIC.PRODUCTS.RATING AS RATING,\n        PUBLIC.PRODUCTS.CREATED_AT AS CREATED_AT\n    FROM PUBLIC.PRODUCTS\n    WHERE PUBLIC.PRODUCTS.CATEGORY = ?1\n) \"PRODUCTS__via__PRODUCT_ID\"\n    ON \"source\".PRODUCT_ID = \"PRODUCTS__via__PRODUCT_ID\".ID [42122-197]"
​​
trace: (107) […]
​​​
[0…99]
0: Array(4) [ "org.h2.message.DbException", "getJdbcSQLException", "DbException.java", … ]
1: Array(4) [ "org.h2.message.DbException", "get", "DbException.java", … ]
2: Array(4) [ "org.h2.message.DbException", "get", "DbException.java", … ]
3: Array(4) [ "org.h2.expression.ExpressionColumn", "optimize", "ExpressionColumn.java", … ]
4: Array(4) [ "org.h2.expression.Alias", "optimize", "Alias.java", … ]
5: Array(4) [ "org.h2.command.dml.Select", "prepare", "Select.java", … ]
6: Array(4) [ "org.h2.command.Parser", "prepare", "Parser.java", … ]
7: Array(4) [ "org.h2.engine.Session", "prepare", "Session.java", … ]
8: Array(4) [ "org.h2.table.TableView", "compileViewQuery", "TableView.java", … ]
9: Array(4) [ "org.h2.table.TableView", "initColumnsAndTables", "TableView.java", … ]
10: Array(4) [ "org.h2.table.TableView", "init", "TableView.java", … ]
11: Array(4) [ "org.h2.table.TableView", "<init>", "TableView.java", … ]
12: Array(4) [ "org.h2.table.TableView", "createTempView", "TableView.java", … ]
13: Array(4) [ "org.h2.command.Parser", "readTableFilter", "Parser.java", … ]
14: Array(4) [ "org.h2.command.Parser", "parseSelectSimpleFromPart", "Parser.java", … ]
15: Array(4) [ "org.h2.command.Parser", "parseSelectSimple", "Parser.java", … ]
16: Array(4) [ "org.h2.command.Parser", "parseSelectSub", "Parser.java", … ]
17: Array(4) [ "org.h2.command.Parser", "parseSelectUnion", "Parser.java", … ]
18: Array(4) [ "org.h2.command.Parser", "parseSelect", "Parser.java", … ]
19: Array(4) [ "org.h2.command.Parser", "parsePrepared", "Parser.java", … ]
20: Array(4) [ "org.h2.command.Parser", "parse", "Parser.java", … ]
21: Array(4) [ "org.h2.command.Parser", "parse", "Parser.java", … ]
22: Array(4) [ "org.h2.command.Parser", "prepareCommand", "Parser.java", … ]
23: Array(4) [ "org.h2.engine.Session", "prepareLocal", "Session.java", … ]
24: Array(4) [ "org.h2.engine.Session", "prepareCommand", "Session.java", … ]
25: Array(4) [ "org.h2.jdbc.JdbcConnection", "prepareCommand", "JdbcConnection.java", … ]
26: Array(4) [ "org.h2.jdbc.JdbcPreparedStatement", "<init>", "JdbcPreparedStatement.java", … ]
27: Array(4) [ "org.h2.jdbc.JdbcConnection", "prepareStatement", "JdbcConnection.java", … ]
28: Array(4) [ "com.mchange.v2.c3p0.impl.NewProxyConnection", "prepareStatement", "NewProxyConnection.java", … ]
29: Array(4) [ "metabase.driver.sql_jdbc.execute$fn__36285", "invokeStatic", "execute.clj", … ]
30: Array(4) [ "metabase.driver.sql_jdbc.execute$fn__36285", "invoke", "execute.clj", … ]
31: Array(4) [ "clojure.lang.MultiFn", "invoke", "MultiFn.java", … ]
32: Array(4) [ "metabase.driver.sql_jdbc.execute$prepared_statement_STAR_", "invokeStatic", "execute.clj", … ]
33: Array(4) [ "metabase.driver.sql_jdbc.execute$prepared_statement_STAR_", "invoke", "execute.clj", … ]
34: Array(4) [ "metabase.driver.sql_jdbc.execute$execute_reducible_query", "invokeStatic", "execute.clj", … ]
35: Array(4) [ "metabase.driver.sql_jdbc.execute$execute_reducible_query", "invoke", "execute.clj", … ]
36: Array(4) [ "metabase.driver.sql_jdbc.execute$execute_reducible_query", "invokeStatic", "execute.clj", … ]
37: Array(4) [ "metabase.driver.sql_jdbc.execute$execute_reducible_query", "invoke", "execute.clj", … ]
38: Array(4) [ "metabase.driver.sql_jdbc$fn__83124", "invokeStatic", "sql_jdbc.clj", … ]
39: Array(4) [ "metabase.driver.sql_jdbc$fn__83124", "invoke", "sql_jdbc.clj", … ]
40: Array(4) [ "metabase.driver.h2$fn__81928", "invokeStatic", "h2.clj", … ]
41: Array(4) [ "metabase.driver.h2$fn__81928", "invoke", "h2.clj", … ]
42: Array(4) [ "clojure.lang.MultiFn", "invoke", "MultiFn.java", … ]
43: Array(4) [ "metabase.query_processor.context$executef", "invokeStatic", "context.clj", … ]
44: Array(4) [ "metabase.query_processor.context$executef", "invoke", "context.clj", … ]
45: Array(4) [ "metabase.query_processor.context.default$default_runf", "invokeStatic", "default.clj", … ]
46: Array(4) [ "metabase.query_processor.context.default$default_runf", "invoke", "default.clj", … ]
47: Array(4) [ "metabase.query_processor.context$runf", "invokeStatic", "context.clj", … ]
48: Array(4) [ "metabase.query_processor.context$runf", "invoke", "context.clj", … ]
49: Array(4) [ "metabase.query_processor.reducible$pivot", "invokeStatic", "reducible.clj", … ]
50: Array(4) [ "metabase.query_processor.reducible$pivot", "invoke", "reducible.clj", … ]
51: Array(4) [ "metabase.query_processor.middleware.mbql_to_native$mbql__GT_native$fn__47089", "invoke", "mbql_to_native.clj", … ]
52: Array(4) [ "metabase.query_processor.middleware.check_features$check_features$fn__46365", "invoke", "check_features.clj", … ]
53: Array(4) [ "metabase.query_processor.middleware.limit$limit$fn__47075", "invoke", "limit.clj", … ]
54: Array(4) [ "metabase.query_processor.middleware.cache$maybe_return_cached_results$fn__46016", "invoke", "cache.clj", … ]
55: Array(4) [ "metabase.query_processor.middleware.optimize_datetime_filters$optimize_datetime_filters$fn__47254", "invoke", "optimize_datetime_filters.clj", … ]
56: Array(4) [ "metabase.query_processor.middleware.auto_parse_filter_values$auto_parse_filter_values$fn__45162", "invoke", "auto_parse_filter_values.clj", … ]
57: Array(4) [ "metabase.query_processor.middleware.wrap_value_literals$wrap_value_literals$fn__40678", "invoke", "wrap_value_literals.clj", … ]
58: Array(4) [ "metabase.query_processor.middleware.annotate$add_column_info$fn__40563", "invoke", "annotate.clj", … ]
59: Array(4) [ "metabase.query_processor.middleware.permissions$check_query_permissions$fn__46240", "invoke", "permissions.clj", … ]
60: Array(4) [ "metabase.query_processor.middleware.pre_alias_aggregations$pre_alias_aggregations$fn__47777", "invoke", "pre_alias_aggregations.clj", … ]
61: Array(4) [ "metabase.query_processor.middleware.cumulative_aggregations$handle_cumulative_aggregations$fn__46438", "invoke", "cumulative_aggregations.clj", … ]
62: Array(4) [ "metabase_enterprise.sandbox.query_processor.middleware.row_level_restrictions$apply_row_level_permissions$fn__49349", "invoke", "row_level_restrictions.clj", … ]
63: Array(4) [ "metabase.query_processor.middleware.resolve_joined_fields$resolve_joined_fields$fn__48090", "invoke", "resolve_joined_fields.clj", … ]
64: Array(4) [ "metabase.query_processor.middleware.resolve_joins$resolve_joins$fn__48395", "invoke", "resolve_joins.clj", … ]
65: Array(4) [ "metabase.query_processor.middleware.add_implicit_joins$add_implicit_joins$fn__44752", "invoke", "add_implicit_joins.clj", … ]
66: Array(4) [ "metabase.query_processor.middleware.large_int_id$convert_id_to_string$fn__47050", "invoke", "large_int_id.clj", … ]
67: Array(4) [ "metabase.query_processor.middleware.format_rows$format_rows$fn__47030", "invoke", "format_rows.clj", … ]
68: Array(4) [ "metabase.query_processor.middleware.desugar$desugar$fn__46504", "invoke", "desugar.clj", … ]
69: Array(4) [ "metabase.query_processor.middleware.binning$update_binning_strategy$fn__45522", "invoke", "binning.clj", … ]
70: Array(4) [ "metabase.query_processor.middleware.resolve_fields$resolve_fields$fn__46040", "invoke", "resolve_fields.clj", … ]
71: Array(4) [ "metabase.query_processor.middleware.add_dimension_projections$add_remapping$fn__44382", "invoke", "add_dimension_projections.clj", … ]
72: Array(4) [ "metabase.query_processor.middleware.add_implicit_clauses$add_implicit_clauses$fn__44613", "invoke", "add_implicit_clauses.clj", … ]
73: Array(4) [ "metabase_enterprise.sandbox.query_processor.middleware.row_level_restrictions$apply_row_level_permissions$fn__49349", "invoke", "row_level_restrictions.clj", … ]
74: Array(4) [ "metabase.query_processor.middleware.upgrade_field_literals$upgrade_field_literals$fn__48825", "invoke", "upgrade_field_literals.clj", … ]
75: Array(4) [ "metabase.query_processor.middleware.add_source_metadata$add_source_metadata_for_source_queries$fn__44909", "invoke", "add_source_metadata.clj", … ]
76: Array(4) [ "metabase_enterprise.sandbox.query_processor.middleware.column_level_perms_check$maybe_apply_column_level_perms_check$fn__48866", "invoke", "column_level_perms_check.clj", … ]
77: Array(4) [ "metabase.query_processor.middleware.reconcile_breakout_and_order_by_bucketing$reconcile_breakout_and_order_by_bucketing$fn__47974", "invoke", "reconcile_breakout_and_order_by_bucketing.clj", … ]
78: Array(4) [ "metabase.query_processor.middleware.auto_bucket_datetimes$auto_bucket_datetimes$fn__45109", "invoke", "auto_bucket_datetimes.clj", … ]
79: Array(4) [ "metabase.query_processor.middleware.resolve_source_table$resolve_source_tables$fn__46087", "invoke", "resolve_source_table.clj", … ]
80: Array(4) [ "metabase.query_processor.middleware.parameters$substitute_parameters$fn__47759", "invoke", "parameters.clj", … ]
81: Array(4) [ "metabase.query_processor.middleware.resolve_referenced$resolve_referenced_card_resources$fn__46139", "invoke", "resolve_referenced.clj", … ]
82: Array(4) [ "metabase.query_processor.middleware.expand_macros$expand_macros$fn__46760", "invoke", "expand_macros.clj", … ]
83: Array(4) [ "metabase.query_processor.middleware.add_timezone_info$add_timezone_info$fn__44918", "invoke", "add_timezone_info.clj", … ]
84: Array(4) [ "metabase.query_processor.middleware.splice_params_in_response$splice_params_in_response$fn__48761", "invoke", "splice_params_in_response.clj", … ]
85: Array(4) [ "metabase.query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__47985$fn__47989", "invoke", "resolve_database_and_driver.clj", … ]
86: Array(4) [ "metabase.driver$do_with_driver", "invokeStatic", "driver.clj", … ]
87: Array(4) [ "metabase.driver$do_with_driver", "invoke", "driver.clj", … ]
88: Array(4) [ "metabase.query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__47985", "invoke", "resolve_database_and_driver.clj", … ]
89: Array(4) [ "metabase.query_processor.middleware.fetch_source_query$resolve_card_id_source_tables$fn__46978", "invoke", "fetch_source_query.clj", … ]
90: Array(4) [ "metabase.query_processor.middleware.store$initialize_store$fn__48770$fn__48771", "invoke", "store.clj", … ]
91: Array(4) [ "metabase.query_processor.store$do_with_store", "invokeStatic", "store.clj", … ]
92: Array(4) [ "metabase.query_processor.store$do_with_store", "invoke", "store.clj", … ]
93: Array(4) [ "metabase.query_processor.middleware.store$initialize_store$fn__48770", "invoke", "store.clj", … ]
94: Array(4) [ "metabase.query_processor.middleware.validate$validate_query$fn__48832", "invoke", "validate.clj", … ]
95: Array(4) [ "metabase.query_processor.middleware.normalize_query$normalize$fn__47102", "invoke", "normalize_query.clj", … ]
96: Array(4) [ "metabase.query_processor.middleware.add_rows_truncated$add_rows_truncated$fn__44770", "invoke", "add_rows_truncated.clj", … ]
97: Array(4) [ "metabase_enterprise.audit.query_processor.middleware.handle_audit_queries$handle_internal_queries$fn__31310", "invoke", "handle_audit_queries.clj", … ]
98: Array(4) [ "metabase.query_processor.middleware.results_metadata$record_and_return_metadata_BANG_$fn__48746", "invoke", "results_metadata.clj", … ]
99: Array(4) [ "metabase.query_processor.reducible$async_qp$qp_STAR___33118$thunk__33119", "invoke", "reducible.clj", … ]
​​​
[100…106]
100: Array(4) [ "metabase.query_processor.reducible$async_qp$qp_STAR___33118$fn__33121", "invoke", "reducible.clj", … ]
101: Array(4) [ "clojure.core$binding_conveyor_fn$fn__5754", "invoke", "core.clj", … ]
102: Array(4) [ "clojure.lang.AFn", "call", "AFn.java", … ]
103: Array(4) [ "java.util.concurrent.FutureTask", "run", "FutureTask.java", … ]
104: Array(4) [ "java.util.concurrent.ThreadPoolExecutor", "runWorker", "ThreadPoolExecutor.java", … ]
105: Array(4) [ "java.util.concurrent.ThreadPoolExecutor$Worker", "run", "ThreadPoolExecutor.java", … ]
106: Array(4) [ "java.lang.Thread", "run", "Thread.java", … ]
​​​
length: 107
​​​
<prototype>: Array []
​​
via: Array [ {…} ]
​​
<prototype>: Object { … }
​
isCancelled: false
​
status: 500
​
<prototype>: Object { … }
app-main.bundle.js:5:2542198
```

</details>

![image](https://user-images.githubusercontent.com/1447303/109198790-ebd3da80-779e-11eb-9962-88d689f6765b.png)

**Information about your Metabase Installation:**
Metabase 1.38.0.1

Related to ~~#14765~~ #15003 and #12586

**Comment by camsaul** [comment]
Not really sure I'd say this is related to #14765, the only thing they have in common is the fact that they have something to do with Pivot Tables. They're two completely unrelated issues tho

**Comment by camsaul** [comment]
Actually, I'm not sure I understand how this really has anything to do with Pivot Tables at all? If you can't even run the query then of course you can't use a Pivot Tables visualization. But you're not going to be able to use a pie chart either. That doesn't make this a Pie Chart bug

**Comment by camsaul** [comment]
I was confused for a second, the repro instructions as-is work just fine. I guess there's another step in there we're we're supposed to choose "Pivot Table" visualization.

**Comment by flamber** [comment]
@camsaul I've now created an issue specifically for the missing logging #15003
Sorry about the steps. Clearly I missed that part in step 3, but I guessed the title and description made that clear enough.

**Comment by camsaul** [comment]
Fixed by #15001
